screen2.0/src/app/applets/differential-gene-expression/page.tsx

```diff
@@ -102,591 +111,442 @@ export default function DifferentialGeneExpression() {
+      coord_start: slider.min,
+      coord_end: slider.max,
+      cellType: ct1,
+      gene_all_start: 0,
+      gene_all_end: 5000000,
+      gene_pc_start: 0,
+      gene_pc_end: 5000000,
+      rank_ctcf_end: 10.0,
+      rank_ctcf_start: -10.0,
+      rank_dnase_end: 10.0,
+      rank_dnase_start: -10.0,
+      rank_enhancer_end: 10.0,
+      rank_enhancer_start: -10.0,
+      rank_promoter_end: 10.0,
+      rank_promoter_start: -10.0,
+      element_type: null,
+      limit: 25000,
+    },
+    fetchPolicy: "cache-and-network",
+    nextFetchPolicy: "cache-first",
+    client,
+  })
 
-        // round y1, y2
-        let ymin: number = data[data.gene].nearbyDEs.ymin
-        let ymax: number = data[data.gene].nearbyDEs.ymax
-        let min_y: number = 0.0
+  // cell type 2
+  const {
+    loading: loading_ct2,
+    error: error_ct2,
+    data: data_ct2,
+  } = useQuery(ZSCORE_QUERY, {
+    variables: {
+      assembly: assembly,
+      coord_chrom: "chr11",
+      coord_start: slider.min,
+      coord_end: slider.max,
+      cellType: ct2,
+      gene_all_start: 0,
+      gene_all_end: 5000000,
+      gene_pc_start: 0,
+      gene_pc_end: 5000000,
+      rank_ctcf_end: 10.0,
+      rank_ctcf_start: -10.0,
+      rank_dnase_end: 10.0,
+      rank_dnase_start: -10.0,
+      rank_enhancer_end: 10.0,
+      rank_enhancer_start: -10.0,
+      rank_promoter_end: 10.0,
+      rank_promoter_start: -10.0,
+      element_type: null,
+      limit: 25000,
+    },
+    fetchPolicy: "cache-and-network",
+    nextFetchPolicy: "cache-first",
+    client,
+  })
 
-        if (ymin < 0) min_y = parseInt(ymin.toString()[0] + (ymin - 0.5).toString()[1]) - 0.5
-        else min_y = parseInt((ymin - 0.5).toString()[0]) - 0.5
-        let max_y: number = parseInt((ymax + 0.5).toString()[0]) + 0.5
+  // server
+  // const data1 = await fetchServer("https://screen-beta-api.wenglab.org/dews/search", payload)
+  // const cellInfo = await getCellInfo()
+  // const cellTypes2 = await getCellTypes()
 
-        if (max_y > 0.5 + ymax) max_y -= 0.5
-        if (min_y < ymin - 0.5) min_y += 0.5
+  const toggleDrawer = (open) => (event) => {
+    if (event.type === "keydown" && (event.key === "Tab" || event.key === "Shift")) {
+      return
+    }
+    setState(open)
+  }
 
-        setMin(min_x)
-        setMax(max_x)
-        setSlider([min_x, max_x])
-        setRange({
-          x: {
-            start: min_x,
-            end: max_x,
-          },
-          y: {
-            start: min_y,
-            end: max_y,
-          },
-        })
+  // set drawer height based on screen size
+  const drawerWidth: number = 550
+  let drawerHeight: number = window.screen.height
+  let drawerHeightTab: number = window.screen.height
 
-        setLoadingChart(false)
-      })
-      .catch((error: Error) => {
-        // logging
-        // throw error
-        return <ErrorMessage error={error} />
-      })
-    setLoadingChart(true)
-  }, [ct1, ct2, gene])
+  // 1080
```

Comment:
What is the reason that this is all being done manually? Setting sizes of elements like this will inherently make screen size responsiveness harder to implement. This is for the collapsible side panel right?

Reply:
yeah, if the height isn't set then you can't scroll down and see all the options. there might be a more creative way of doing this but right now its probably good. this is also where we can implement a way to make the hamburger menu attach to the top of the screen